seqr/views/utils/airtable_utils.py

```diff
@@ -44,15 +47,47 @@ def safe_create_record(self, record_type, record):
         except Exception as e:
             logger.error(f'Airtable create "{record_type}" error: {e}', self._user)
 
-    def fetch_records(self, record_type, fields, or_filters):
-        self._session.params.update({'fields[]': fields, 'pageSize': PAGE_SIZE})
+    def safe_patch_records(self, record_type, record_or_filters, record_and_filters, update, max_records=PAGE_SIZE - 1):
+        try:
+            self._patch_record(record_type, record_or_filters, record_and_filters, update, max_records)
+        except Exception as e:
+            logger.error(f'Airtable patch "{record_type}" error: {e}', self._user, detail={
+                'or_filters': record_or_filters, 'and_filters': record_and_filters, 'update': update,
+            })
+
+    def _patch_record(self, record_type, record_or_filters, record_and_filters, update, max_records):
+        records = self.fetch_records(
+            record_type, fields=record_or_filters.keys(), or_filters=record_or_filters, and_filters=record_and_filters,
+            page_size=max_records+1,
+        )
+        if not records or len(records) > max_records:
+            raise ValueError('Unable to identify record to update')
+
+        self._session.params = {}
+        errors = []
+        for record_id in records.keys():
+            response = self._session.patch(f'{self._url}/{record_type}/{record_id}', json={'fields': update})
```

Comment:
Definitely a nit, but is there a reason not to put the `patch()` call itself inside the `try/except`?  I think the `patch()` on its own can throw connection errors/timeouts etc. we'd want to continue the loop right?